diff --git a/core/modules/responsive_image/responsive_image.module b/core/modules/responsive_image/responsive_image.module
index 85a2ff94ddd..58c980ba2ba 100644
--- a/core/modules/responsive_image/responsive_image.module
+++ b/core/modules/responsive_image/responsive_image.module
@@ -525,13 +525,17 @@ function _responsive_image_image_style_url($style_name, $path) {
   /** @var \Drupal\Core\File\FileUrlGeneratorInterface $file_url_generator */
   $file_url_generator = \Drupal::service('file_url_generator');
 
-  $entity = ImageStyle::load($style_name);
-  if ($entity instanceof ImageStyle) {
-    return $file_url_generator->transformRelative($entity->buildUrl($path));
+  if ($style = ImageStyle::load($style_name)) {
+    // Generate absolute URL to the image style derivative.
+    $derivative_uri = $style->buildUri($path);
+    return $file_url_generator->generateAbsoluteString($derivative_uri);
   }
-  return $file_url_generator->generateString($path);
+
+  // Fallback to absolute URL of original file.
+  return $file_url_generator->generateAbsoluteString($path);
 }
 
+
 /**
  * Implements hook_library_info_alter().
  *
